SwiftUI/Model/SearchOperation/SearchOperation.mm

```diff
@@ -49,29 +49,31 @@ - (id)initWithSearchText:(NSString *)searchText zimFileIDs:(NSSet *)zimFileIDs {
 
 /// Perform index and title based searches.
 - (void)performSearch {
-    // get a list of archives that are included in search
-    typedef std::unordered_map<std::string, zim::Archive> archives_map;
-    auto *allArchives = static_cast<archives_map *>([[ZimFileService sharedInstance] getArchives]);
-    
-    std::vector<zim::Archive> indexSearchArchives = std::vector<zim::Archive>();
-    std::vector<zim::Archive> titleSearchArchives = std::vector<zim::Archive>();
-    for (NSUUID *zimFileID in self.zimFileIDs) {
-        std::string zimFileID_C = [[[zimFileID UUIDString] lowercaseString] cStringUsingEncoding:NSUTF8StringEncoding];
-        try {
-            auto archive = allArchives->at(zimFileID_C);
-            if (archive.hasFulltextIndex()) {
-                indexSearchArchives.push_back(archive);
-            }
-            titleSearchArchives.push_back(archive);
-        } catch (std::out_of_range) { }
-    }
-    
-    // perform index and title search
-    [self addIndexSearchResults:indexSearchArchives];
-    if (titleSearchArchives.size() > 0) {
-        int count = std::max((35 - (int)[self.results count]) / (int)titleSearchArchives.size(), 5);
-        [self addTitleSearchResults:titleSearchArchives count:(int)count];
-    }
+    try {
```

Comment:
You probably don't want to try/catch the whole loop adding the archive but try/catch each adding of an archive.

Else, the first corrupted zim file with stop the loop.

Reply:
Thanks, make sense. Please see it updated.